edit/edit.go

```diff
@@ -136,21 +136,29 @@ func InterpretLabelForWorkspaceLocation(root string, target string) (buildFile s
 	}
 
 	if strings.HasPrefix(target, "//") {
-		buildFile = path.Join(rootDir, pkg, "BUILD")
+		buildFile = filepath.Join(rootDir, pkg, "BUILD")
+		if !isFile(buildFile) {
+			// try it with the .bazel extension
+			buildFile += ".bazel"
+		}
 		return
 	}
 	if isFile(pkg) {
 		// allow operation on other files like WORKSPACE
 		buildFile = pkg
-		pkg = path.Join(relativePath, filepath.Dir(pkg))
+		pkg = filepath.Join(relativePath, filepath.Dir(pkg))
 		return
 	}
 	if pkg != "" {
-		buildFile = pkg + "/BUILD"
+		buildFile = filepath.Join(pkg, "/BUILD")
```

Comment:
Good catch, but now the tests fail on Windows, you probably need to update them by replacing a slash with `filepath.Separator`.

Reply:
Also, shouldn't it be `"BUILD"` instead of `"/BUILD"`?

Reply:
I only need to remove the `/` because `Join` already adds that. (I don't have windows so I only notice it from the buildkite run. Getting this to pass on windows is a bit tricky, I'll notify you again once I have it figured out.

I think the gist of it is that `ParseLabel` does not work for absolute windows paths. It was only tested on unix-style paths: https://github.com/bazelbuild/buildtools/blob/96b69a8e017b0943fa3003d118794324e2b75344/edit/edit_test.go#L46. The reason is that the code simply splits on the `:`, which is a part of the path on windows.

Reply:
That's probably the reason, at least it can explain the following error: https://storage.googleapis.com/bazel-untrusted-buildkite-artifacts/bf49dbeb-eb10-4ee6-bbdc-669c127fbc3f/edit%5Cgo_default_test%5Cattempt_1.log

Feel free to just revert the change to this line if you don't want to fix it now, it's not directly relevant to your pull request.

Reply:
Another suggestion is to ignore this issue for now, remove the last two test cases, and create a new issue for fixing the parsing issues on windows.

Reply:
I removed the unrelated change and also removed the test cases (that I added) which are failing on windows. I can add a new ticket for these issues afterwards if you'd like that.

Also: please let me know if I should rebase/squash this once you're done with the review. I'm not familiar with how you handle that on this project.